protocols/identify/src/behaviour.rs

```diff
@@ -670,6 +685,39 @@ impl PeerCache {
     }
 }
 
+#[derive(Debug, Clone)]
+#[allow(clippy::large_enum_variant)]
+pub enum CryptoKey {
```

Comment:
Since I dont believe this would be used outside of the crate, unlesss `CryptoKey` is expected to be used in some manner outside of the crate too.

```suggestion
pub(crate) enum CryptoKey {
```

Reply:
That would trigger a visibility warning because `Handler`'s public constructor uses this type, though I don't expect the constructor can actually be called outside of the crate.

Reply:
Hmm then we could also mark that as `pub(crate)`?